File: Rude-Chat-flask-agents/MCP-Server/MCPServer/Models/ChatModels.cs
using System.Text.Json.Serialization;
using Newtonsoft.Json;

namespace MCPServer.Models;

public class ChatMessage
{
    [JsonPropertyName("id")]
    [JsonProperty("id")]
    public string Id { get; set; } = Guid.NewGuid().ToString();
    
    [JsonPropertyName("userId")]
    [JsonProperty("userId")]
    public string UserId { get; set; } = string.Empty;
    
    [JsonPropertyName("sessionId")]
    [JsonProperty("sessionId")]
    public string SessionId { get; set; } = string.Empty;
    
    [JsonPropertyName("role")]
    [JsonProperty("role")]
    public string Role { get; set; } = string.Empty; // "user" or "assistant"
    
    [JsonPropertyName("content")]
    [JsonProperty("content")]
    public string Content { get; set; } = string.Empty;
    
    [JsonPropertyName("timestamp")]
    [JsonProperty("timestamp")]
    public DateTime Timestamp { get; set; } = DateTime.UtcNow;
    
    [JsonPropertyName("parentMessageId")]
    [JsonProperty("parentMessageId")]
    public string? ParentMessageId { get; set; }
    
    [JsonPropertyName("metadata")]
    [JsonProperty("metadata")]
    public Dictionary<string, object>? Metadata { get; set; }
    
    [JsonPropertyName("_partitionKey")]
    [JsonProperty("_partitionKey")]
    public string PartitionKey => UserId;
}

public class ChatSession
{
    [JsonPropertyName("id")]
    [JsonProperty("id")]
    public string Id { get; set; } = Guid.NewGuid().ToString();
    
    [JsonPropertyName("userId")]
    [JsonProperty("userId")]
    public string UserId { get; set; } = string.Empty;
    
    [JsonPropertyName("title")]
    [JsonProperty("title")]
    public string Title { get; set; } = string.Empty;
    
    [JsonPropertyName("createdAt")]
    [JsonProperty("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    [JsonPropertyName("lastMessageAt")]
    [JsonProperty("lastMessageAt")]
    public DateTime LastMessageAt { get; set; } = DateTime.UtcNow;
    
    [JsonPropertyName("messageCount")]
    [JsonProperty("messageCount")]
    public int MessageCount { get; set; } = 0;
      [JsonPropertyName("isArchived")]
    [JsonProperty("isArchived")]
    public bool IsArchived { get; set; } = false;
    
    [JsonPropertyName("documents")]
    [JsonProperty("documents")]
    public List<SessionDocument> Documents { get; set; } = new();
    
    [JsonPropertyName("_partitionKey")]
    [JsonProperty("_partitionKey")]
    public string PartitionKey => UserId;
}

public class ChatHistoryRequest
{
    public string UserId { get; set; } = string.Empty;
    public string SessionId { get; set; } = string.Empty;
    public int PageSize { get; set; } = 50;
    public string? ContinuationToken { get; set; }
}

public class ChatHistoryResponse
{
    public List<ChatMessage> Messages { get; set; } = new();
    public string? ContinuationToken { get; set; }
    public bool HasMore { get; set; }
    public int TotalCount { get; set; }
}

public class SessionListRequest
{
    public string UserId { get; set; } = string.Empty;
    public int PageSize { get; set; } = 20;
    public string? ContinuationToken { get; set; }
    public bool IncludeArchived { get; set; } = false;
}

public class SessionListResponse
{
    public List<ChatSession> Sessions { get; set; } = new();
    public string? ContinuationToken { get; set; }    public bool HasMore { get; set; }
}

public class SessionDocument
{
    [JsonPropertyName("documentId")]
    [JsonProperty("documentId")]
    public string DocumentId { get; set; } = string.Empty;
    
    [JsonPropertyName("fileName")]
    [JsonProperty("fileName")]
    public string FileName { get; set; } = string.Empty;
    
    [JsonPropertyName("fileSize")]
    [JsonProperty("fileSize")]
    public long FileSize { get; set; }
    
    [JsonPropertyName("blobUrl")]
    [JsonProperty("blobUrl")]
    public string BlobUrl { get; set; } = string.Empty;
    
    [JsonPropertyName("uploadDate")]
    [JsonProperty("uploadDate")]
    public DateTime UploadDate { get; set; } = DateTime.UtcNow;
    
    [JsonPropertyName("status")]
    [JsonProperty("status")]
    public int Status { get; set; } // DocumentStatus enum value
}

public class ChatCompletionRequest
{
    [JsonPropertyName("messages")]
    [JsonProperty("messages")]
    public List<ChatMessage> Messages { get; set; } = new();
    
    [JsonPropertyName("userId")]
    [JsonProperty("userId")]
    public string UserId { get; set; } = string.Empty;
    
    [JsonPropertyName("sessionId")]
    [JsonProperty("sessionId")]
    public string SessionId { get; set; } = string.Empty;
    
    [JsonPropertyName("useRAG")]
    [JsonProperty("useRAG")]
    public bool UseRAG { get; set; } = false;
    
    [JsonPropertyName("useMCPTools")]
    [JsonProperty("useMCPTools")]
    public bool UseMCPTools { get; set; } = false;
}

public class ChatCompletionResponse
{
    [JsonPropertyName("message")]
    [JsonProperty("message")]
    public ChatMessage Message { get; set; } = new();
    
    [JsonPropertyName("sources")]
    [JsonProperty("sources")]
    public List<object>? Sources { get; set; }
    
    [JsonPropertyName("toolCalls")]
    [JsonProperty("toolCalls")]
    public List<object>? ToolCalls { get; set; }
    
    [JsonPropertyName("agentInteractions")]
    [JsonProperty("agentInteractions")]
    public List<AgentInteraction>? AgentInteractions { get; set; }
}

public class AgentInteraction
{
    [JsonPropertyName("id")]
    [JsonProperty("id")]
    public string Id { get; set; } = Guid.NewGuid().ToString();
    
    [JsonPropertyName("agentName")]
    [JsonProperty("agentName")]
    public string AgentName { get; set; } = string.Empty;
    
    [JsonPropertyName("action")]
    [JsonProperty("action")]
    public string Action { get; set; } = string.Empty;
    
    [JsonPropertyName("result")]
    [JsonProperty("result")]
    public string Result { get; set; } = string.Empty;
    
    [JsonPropertyName("timestamp")]
    [JsonProperty("timestamp")]
    public DateTime Timestamp { get; set; } = DateTime.UtcNow;
    
    [JsonPropertyName("status")]
    [JsonProperty("status")]
    public string Status { get; set; } = "success"; // success, error, in-progress
    
    [JsonPropertyName("duration")]
    [JsonProperty("duration")]
    public TimeSpan? Duration { get; set; }
}

File: Rude-Chat-flask-agents/Python-MCP-Server/src/models/chat_models.py
from dataclasses import dataclass, field
from typing import List, Optional, Dict, Any
from datetime import datetime
import uuid

@dataclass
class ChatMessage:
    id: str = field(default_factory=lambda: str(uuid.uuid4()))
    role: str = ""  # "user" or "assistant"
    content: Optional[str] = None
    timestamp: datetime = field(default_factory=datetime.utcnow)
    session_id: str = ""
    user_id: str = ""

@dataclass
class ChatCompletionRequest:
    messages: List[ChatMessage]
    session_id: str
    user_id: str
    max_tokens: Optional[int] = None
    temperature: Optional[float] = None

@dataclass
class ChatCompletionResponse:
    message: ChatMessage
    agent_interactions: List['AgentInteraction'] = field(default_factory=list)

@dataclass
class AgentInteraction:
    id: str = field(default_factory=lambda: str(uuid.uuid4()))
    agent_name: str = ""
    action: str = ""
    details: str = ""
    status: str = ""  # "in-progress", "success", "error"
    timestamp: datetime = field(default_factory=datetime.utcnow)
    duration: Optional[float] = None  # in seconds

@dataclass
class ChatSession:
    id: str = field(default_factory=lambda: str(uuid.uuid4()))
    user_id: str = ""
    title: str = ""
    created_at: datetime = field(default_factory=datetime.utcnow)
    last_message_at: Optional[datetime] = None
    documents: List['DocumentMetadata'] = field(default_factory=list)
    is_archived: bool = False

@dataclass
class DocumentMetadata:
    id: str = field(default_factory=lambda: str(uuid.uuid4()))
    filename: str = ""
    blob_url: str = ""
    size: int = 0
    content_type: str = ""
    upload_timestamp: datetime = field(default_factory=datetime.utcnow)
    processing_status: int = 0  # 0=uploading, 1=processing, 2=indexed
    user_id: str = ""
    session_id: str = ""

@dataclass
class ChatHistoryRequest:
    user_id: str
    session_id: str
    page_size: int = 50
    continuation_token: Optional[str] = None

@dataclass
class ChatHistoryResponse:
    messages: List[ChatMessage]
    continuation_token: Optional[str] = None
    has_more: bool = False

@dataclass
class SessionListRequest:
    user_id: str
    page_size: int = 20
    continuation_token: Optional[str] = None
    include_archived: bool = False

@dataclass
class SessionListResponse:
    sessions: List[ChatSession]
    continuation_token: Optional[str] = None
    has_more: bool = False

@dataclass
class CreateSessionRequest:
    user_id: str
    title: str = ""

@dataclass
class DocumentSearchRequest:
    query: str
    user_id: str
    session_id: str
    top_k: int = 5

@dataclass
class DocumentSearchResult:
    id: str
    filename: str
    content: str
    score: float
    metadata: Dict[str, Any] = field(default_factory=dict)

@dataclass
class McpServerInfo:
    name: str = "Python Azure MCP Server"
    version: str = "1.0.0"
    capabilities: 'McpCapabilities' = field(default_factory=lambda: McpCapabilities())

@dataclass
class McpCapabilities:
    tools: 'McpToolCapabilities' = field(default_factory=lambda: McpToolCapabilities())

@dataclass
class McpToolCapabilities:
    list_changed: bool = False

@dataclass
class McpTool:
    name: str
    description: str
    input_schema: Dict[str, Any] = field(default_factory=dict)

@dataclass
class McpToolCall:
    name: str
    arguments: Dict[str, Any] = field(default_factory=dict)

@dataclass
class McpToolResult:
    content: List[Dict[str, Any]] = field(default_factory=list)
    is_error: bool = False

@dataclass
class AgentStatus:
    agent_id: str
    name: str
    description: str
    domains: List[str]
    health: 'AgentHealth' = field(default_factory=lambda: AgentHealth())

@dataclass
class AgentHealth:
    agent_id: str = ""
    status: str = "healthy"  # "healthy", "degraded", "unhealthy"
    last_check: datetime = field(default_factory=datetime.utcnow)
    response_time_ms: float = 0.0
    success_rate: float = 100.0
    error_message: Optional[str] = None
